{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "import geopandas as gpd\n",
    "import pandas as pd\n",
    "import classes.entropycalculator as ec\n",
    "from spatialentropy import altieri_entropy, leibovici_entropy\n",
    "from scipy.stats import entropy\n",
    "import shapely\n",
    "import numpy as np\n",
    "import gc\n",
    "\n",
    "\n",
    "from tqdm import tqdm\n",
    "tqdm.pandas()\n",
    "\n",
    "# select the part to handle\n",
    "part = 3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "gemeenten = gpd.read_parquet(f\"data/buurten_parts/buurten_{part}.parquet\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "def _get_shannon_entropy(labels, base=2):\n",
    "    # get the total count of the labels\n",
    "    total_count = len(labels)\n",
    "    # get the unique labels and their counts\n",
    "    _, label_counts = np.unique(labels, return_counts=True)\n",
    "\n",
    "    probs = label_counts / total_count\n",
    "    # get the entropy\n",
    "    return entropy(probs, base=base)\n",
    "\n",
    "def gm_total_amenities_entropy(gm_name, filter_i, area):\n",
    "    L0_BLACKLIST, L1_BLACKLIST = ec.getfilter(filter_i)\n",
    "    amenity_gdf = gpd.read_parquet(f\"data/gm_amenities/amenities_{gm_name}.parquet\")\n",
    "    \n",
    "    amenity_gdf = amenity_gdf[amenity_gdf.within(area)] \n",
    "    # apply filters\n",
    "    amenity_gdf = amenity_gdf[~amenity_gdf.L0_category.isin(L0_BLACKLIST)]\n",
    "    if L1_BLACKLIST:\n",
    "        for key, value in L1_BLACKLIST.items():\n",
    "            amenity_gdf = amenity_gdf[\n",
    "                ~(\n",
    "                    (amenity_gdf.L0_category == key)\n",
    "                    & (amenity_gdf.L1_category.isin(value))\n",
    "                )\n",
    "            ]\n",
    "    \n",
    "    # total number of amenities\n",
    "    total_amenities = len(amenity_gdf)\n",
    "    \n",
    "    points = [[point.x, point.y] for point in amenity_gdf.geometry]\n",
    "    \n",
    "    # calculate entropy\n",
    "    L0 = amenity_gdf.loc[:, f\"L0_category\"].values\n",
    "    L1 = amenity_gdf.loc[:, f\"L1_category\"].values\n",
    "    try:\n",
    "        L0_entropy_lei = leibovici_entropy(points, L0, base=2).entropy\n",
    "        L0_entropy_alt = altieri_entropy(points, L0, base=2).entropy\n",
    "    except IndexError:\n",
    "        L0_entropy_lei = 0\n",
    "        L0_entropy_alt = 0\n",
    "    try:\n",
    "        L1_entropy_lei = leibovici_entropy(points, L1, base=2).entropy\n",
    "        L1_entropy_alt = altieri_entropy(points, L1, base=2).entropy\n",
    "    except IndexError:\n",
    "        L1_entropy_lei = 0\n",
    "        L1_entropy_alt = 0\n",
    "    del points\n",
    "    gc.collect()\n",
    "    \n",
    "    return total_amenities, L0_entropy_lei, L1_entropy_lei, L0_entropy_alt, L1_entropy_alt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filter 0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 20%|█▉        | 277/1394 [1:35:22<5:37:46, 18.14s/it]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 20%|██        | 281/1394 [1:36:34<5:36:18, 18.13s/it]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 29%|██▊       | 400/1394 [2:09:09<08:59,  1.84it/s]  /home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 33%|███▎      | 466/1394 [2:09:29<04:00,  3.86it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 44%|████▍     | 610/1394 [2:11:06<09:27,  1.38it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 56%|█████▋    | 786/1394 [2:15:02<14:48,  1.46s/it]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 63%|██████▎   | 884/1394 [2:15:45<06:13,  1.36it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 75%|███████▌  | 1047/1394 [2:17:17<01:36,  3.59it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 79%|███████▉  | 1105/1394 [2:17:34<01:26,  3.36it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 82%|████████▏ | 1141/1394 [2:17:44<00:54,  4.62it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 82%|████████▏ | 1145/1394 [2:17:45<00:53,  4.69it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 83%|████████▎ | 1156/1394 [2:17:47<00:51,  4.64it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 83%|████████▎ | 1158/1394 [2:17:47<00:51,  4.58it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 84%|████████▍ | 1174/1394 [2:17:51<00:47,  4.65it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 88%|████████▊ | 1223/1394 [2:18:33<01:05,  2.60it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "100%|██████████| 1394/1394 [2:19:30<00:00,  6.00s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filter 1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 20%|██        | 281/1394 [1:24:16<5:32:32, 17.93s/it]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 20%|██        | 284/1394 [1:25:10<5:31:15, 17.91s/it]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 23%|██▎       | 325/1394 [1:37:23<5:18:47, 17.89s/it]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 23%|██▎       | 327/1394 [1:38:00<5:22:11, 18.12s/it]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 24%|██▍       | 339/1394 [1:41:35<5:13:37, 17.84s/it]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 27%|██▋       | 372/1394 [1:51:24<4:56:19, 17.40s/it]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 29%|██▉       | 402/1394 [1:56:05<06:51,  2.41it/s]  /home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 32%|███▏      | 444/1394 [1:56:18<04:28,  3.54it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 33%|███▎      | 457/1394 [1:56:21<04:26,  3.51it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 35%|███▍      | 484/1394 [1:56:29<04:28,  3.39it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 39%|███▉      | 543/1394 [1:57:21<05:37,  2.52it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 39%|███▉      | 544/1394 [1:57:21<05:01,  2.82it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 44%|████▍     | 610/1394 [1:57:51<09:22,  1.39it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 56%|█████▌    | 781/1394 [2:01:23<17:43,  1.74s/it]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 56%|█████▋    | 786/1394 [2:01:30<13:04,  1.29s/it]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 58%|█████▊    | 804/1394 [2:01:35<02:32,  3.88it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 58%|█████▊    | 805/1394 [2:01:35<02:32,  3.86it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 63%|██████▎   | 875/1394 [2:02:05<06:16,  1.38it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 63%|██████▎   | 878/1394 [2:02:08<06:12,  1.39it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 63%|██████▎   | 884/1394 [2:02:12<06:07,  1.39it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 64%|██████▍   | 893/1394 [2:02:18<05:58,  1.40it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 64%|██████▍   | 897/1394 [2:02:21<05:54,  1.40it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 65%|██████▍   | 905/1394 [2:02:27<05:50,  1.39it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 70%|██████▉   | 971/1394 [2:03:04<04:14,  1.66it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 73%|███████▎  | 1021/1394 [2:03:34<03:44,  1.66it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 75%|███████▌  | 1047/1394 [2:03:42<01:37,  3.57it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 75%|███████▌  | 1052/1394 [2:03:44<01:34,  3.63it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 79%|███████▊  | 1097/1394 [2:03:56<01:26,  3.44it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 79%|███████▉  | 1102/1394 [2:03:58<01:25,  3.40it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 79%|███████▉  | 1105/1394 [2:03:59<01:24,  3.44it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 81%|████████  | 1128/1394 [2:04:05<01:17,  3.44it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 81%|████████  | 1129/1394 [2:04:06<01:17,  3.43it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 82%|████████▏ | 1141/1394 [2:04:08<00:56,  4.49it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 83%|████████▎ | 1154/1394 [2:04:11<00:52,  4.55it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 83%|████████▎ | 1158/1394 [2:04:12<00:52,  4.52it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 83%|████████▎ | 1162/1394 [2:04:13<00:50,  4.58it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 84%|████████▍ | 1171/1394 [2:04:15<00:49,  4.51it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 84%|████████▍ | 1174/1394 [2:04:16<00:48,  4.57it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 88%|████████▊ | 1223/1394 [2:04:52<01:09,  2.46it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 88%|████████▊ | 1226/1394 [2:04:53<01:00,  2.77it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 89%|████████▉ | 1238/1394 [2:04:57<00:51,  3.02it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 89%|████████▉ | 1243/1394 [2:04:59<00:50,  2.97it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 90%|█████████ | 1258/1394 [2:05:04<00:47,  2.84it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 96%|█████████▋| 1342/1394 [2:05:32<00:17,  2.97it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 98%|█████████▊| 1361/1394 [2:05:39<00:11,  2.93it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 99%|█████████▉| 1386/1394 [2:05:47<00:02,  3.47it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "100%|██████████| 1394/1394 [2:05:49<00:00,  5.42s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filter 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 20%|██        | 281/1394 [1:24:01<5:37:33, 18.20s/it]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 20%|██        | 284/1394 [1:24:55<5:34:03, 18.06s/it]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 23%|██▎       | 325/1394 [1:37:13<5:21:20, 18.04s/it]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 24%|██▍       | 339/1394 [1:41:24<5:17:03, 18.03s/it]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 27%|██▋       | 372/1394 [1:51:16<5:05:33, 17.94s/it]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 29%|██▉       | 402/1394 [1:56:09<07:08,  2.32it/s]  /home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 32%|███▏      | 444/1394 [1:56:21<04:34,  3.46it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 33%|███▎      | 457/1394 [1:56:25<04:31,  3.45it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 35%|███▍      | 484/1394 [1:56:33<04:35,  3.30it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 39%|███▉      | 543/1394 [1:57:26<05:43,  2.48it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 39%|███▉      | 544/1394 [1:57:26<05:07,  2.76it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 44%|████▍     | 610/1394 [1:57:56<09:33,  1.37it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 56%|█████▌    | 781/1394 [2:01:31<18:02,  1.77s/it]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 56%|█████▋    | 786/1394 [2:01:38<13:20,  1.32s/it]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 58%|█████▊    | 804/1394 [2:01:43<02:40,  3.67it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 58%|█████▊    | 805/1394 [2:01:43<02:39,  3.69it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 63%|██████▎   | 875/1394 [2:02:14<06:16,  1.38it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 63%|██████▎   | 878/1394 [2:02:16<06:16,  1.37it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 63%|██████▎   | 884/1394 [2:02:21<06:12,  1.37it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 64%|██████▍   | 893/1394 [2:02:27<06:12,  1.34it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 64%|██████▍   | 895/1394 [2:02:29<06:07,  1.36it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 64%|██████▍   | 897/1394 [2:02:30<06:02,  1.37it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 65%|██████▍   | 905/1394 [2:02:36<05:57,  1.37it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 70%|██████▉   | 971/1394 [2:03:14<04:18,  1.64it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 73%|███████▎  | 1021/1394 [2:03:44<03:46,  1.64it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 75%|███████▌  | 1047/1394 [2:03:52<01:39,  3.50it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 75%|███████▌  | 1052/1394 [2:03:54<01:36,  3.54it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 79%|███████▊  | 1097/1394 [2:04:06<01:28,  3.36it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 79%|███████▉  | 1102/1394 [2:04:08<01:29,  3.28it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 79%|███████▉  | 1105/1394 [2:04:09<01:26,  3.33it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 81%|████████  | 1128/1394 [2:04:16<01:19,  3.33it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 81%|████████  | 1129/1394 [2:04:16<01:19,  3.33it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 82%|████████▏ | 1141/1394 [2:04:19<00:58,  4.34it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 83%|████████▎ | 1154/1394 [2:04:22<00:54,  4.39it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 83%|████████▎ | 1162/1394 [2:04:24<00:51,  4.48it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 84%|████████▍ | 1171/1394 [2:04:26<00:49,  4.47it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 84%|████████▍ | 1174/1394 [2:04:26<00:49,  4.48it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 88%|████████▊ | 1223/1394 [2:05:03<01:07,  2.54it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 88%|████████▊ | 1226/1394 [2:05:04<01:00,  2.79it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 89%|████████▉ | 1238/1394 [2:05:08<00:52,  2.98it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 89%|████████▉ | 1243/1394 [2:05:09<00:51,  2.91it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 90%|█████████ | 1258/1394 [2:05:15<00:47,  2.85it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      " 96%|█████████▋| 1342/1394 [2:05:43<00:18,  2.88it/s]/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:63: RuntimeWarning: invalid value encountered in divide\n",
      "  w = w / w.sum()\n",
      "/home/jleeuwenburgh/miniconda3/envs/scriptie/lib/python3.8/site-packages/spatialentropy/_altieri_entropy.py:68: RuntimeWarning: invalid value encountered in divide\n",
      "  pz = pz / pz.sum()\n",
      "100%|██████████| 1394/1394 [2:06:01<00:00,  5.42s/it]\n"
     ]
    }
   ],
   "source": [
    "for filter in [0, 1, 2]:\n",
    "    print(f\"Filter {filter}\")\n",
    "    for i, gm in tqdm(gemeenten.iterrows(), total=len(gemeenten)):\n",
    "        total_amenities, L0_entropy_lei, L1_entropy_lei, L0_entropy_alt, L1_entropy_alt = gm_total_amenities_entropy(gm[\"gemeentenaam\"], filter, gm[\"geometry\"])\n",
    "        gemeenten.at[i, f\"total_amenities_{filter}\"] = total_amenities\n",
    "        gemeenten.at[i, f\"L0_leibovici_{filter}\"] = L0_entropy_lei\n",
    "        gemeenten.at[i, f\"L1_leibovici_{filter}\"] = L1_entropy_lei\n",
    "        gemeenten.at[i, f\"L0_altieri_{filter}\"] = L0_entropy_alt\n",
    "        gemeenten.at[i, f\"L1_altieri_{filter}\"] = L1_entropy_alt\n",
    "        \n",
    "        del total_amenities, L0_entropy_lei, L1_entropy_lei\n",
    "        gc.collect()\n",
    "    \n",
    "gemeenten.to_parquet(f\"data/buurten_parts/calculated/buurten_{part}.parquet\")\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "scriptie_38",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.19"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
